src/coord/src/coord.rs

```diff
@@ -2106,14 +2106,40 @@ impl Coordinator {
 
     async fn sequence_create_secret(
         &mut self,
+        session: &Session,
         plan: CreateSecretPlan,
     ) -> Result<ExecuteResponse, CoordError> {
         let CreateSecretPlan {
             name,
-            secret: _,
+            secret,
             if_not_exists,
         } = plan;
 
+        let compute_instance = self
+            .catalog
+            .resolve_compute_instance(session.vars().cluster())?
+            .id;
+
+        let temp_storage = RowArena::new();
+        self.dataflow_builder(compute_instance).prep_scalar_expr(
+            &mut secret.secret_as.clone(),
+            ExprPrepStyle::OneShot {
+                logical_time: None,
+                session,
+            },
+        )?;
+        let evaled = secret.secret_as.eval(&[], &temp_storage)?;
+        let ty = secret.secret_as.typ(&RelationType::empty());
+
+        // TODO martin: hook the payload into a secrets backend
+        let _payload = match ty.scalar_type {
```

Comment:
This type check should be handled by the `type_as` call below (see suggestion!), so that any type that can be coerced to `bytea` can be used. Then you can just `unwrap_bytes()` here; that method already functions as a debug assert that things are in fact bytes.